cmd/cofidectl/cmd/trustzone/helm/helm.go

```diff
@@ -107,19 +116,34 @@ func (c *HelmCommand) overrideValues(ds datasource.DataSource, tzName string, va
 		return err
 	}
 
+	// Make a copy of the cluster to rollback if needed.
+	oldCluster := proto.Clone(cluster).(*clusterpb.Cluster)
+
 	cluster.ExtraHelmValues, err = structpb.NewStruct(values)
 	if err != nil {
 		return err
 	}
 
-	// Check that the values are acceptable.
-	generator := helm.NewHelmValuesGenerator(trustZone, cluster, ds, nil)
-	if _, err = generator.GenerateValues(); err != nil {
+	_, err = ds.UpdateCluster(cluster)
+	if err != nil {
 		return err
 	}
 
-	_, err = ds.UpdateCluster(cluster)
-	return err
+	// Check that the values are acceptable.
+	_, err = provisionPlugin.GetHelmValues(ctx, ds, &provision.GetHelmValuesOpts{
+		TrustZoneName: tzName,
+		ClusterName:   cluster.GetName(),
+	})
+	if err != nil {
+		slog.Error("Failed to generate Helm values, rolling back", "error", err)
+		// Rollback the cluster to the old state.
+		_, rollbackErr := ds.UpdateCluster(oldCluster)
+		if rollbackErr != nil {
+			return fmt.Errorf("failed to rollback cluster: %w", rollbackErr)
+		}
+		return err
+	}
+	return nil
```

Comment:
Was wondering if we could verify before applying to the ds here, but I guess this is tricky as `GetHelmValues` is a receiver method. 

More generally, do we think these kinds of rollback handling cases are likely for inbound interactions with our datastore (in whatever flavour of driving means: CLI, provider, API etc), and should we think about adding a transactional layer that makes rollback a first class concept?